File: pinyouggou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {

     List<TbBrand> findAll();

    PageResult findPage(int pageNumber, int pageSize);

    void add(TbBrand tbBrand);

    void update(TbBrand tbBrand);

    TbBrand findOne(Long id);

    void delete(Long [] ids);

    PageResult search(/*TbBrandExample example,*/ int pageNum, int pageSize);


    PageResult findPage(TbBrand tbBrand, int pageNumber, int pageSize);

    List<Map> selectBrandOptions();

}

File: pinyouggou_parent/pinyougou_manager_web/src/main/webapp/js/service/brandService.js
//服务层，用户发送请求，与后端进行请求交互
//参数一：服务名称  参数二：服务要处理的事情
app.service("brandService",function ($http) {
    //查询所有品牌列表
    this.findAll = function () {
        return $http.get("../brand/findAll.do");
    }
    //分页查询
    this.findPage = function (pageNumber,pageSize,searchEntity) {
        return  $http.post('../brand/findPage.do?pageNumber='+pageNumber
            +"&pageSize="+pageSize,searchEntity);
    }
    //新增品牌
    this.add=function (entity) {
        return $http.post("../brand/add.do",entity);
    }
    //修改品牌
    this.update=function (entity) {
        return $http.post("../brand/update.do",entity);
    }
    //修改数据回显
    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }
    //批量删除
    this.dele=function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    }
    this.selectBrandOptions=function () {
        return $http.get("../brand/selectBrandOptions.do");

    }

});
